pkg/kube/krt/README.md

```diff
@@ -0,0 +1,157 @@
+# `krt`: **K**ubernetes Declarative Controller **R**un**t**ime
+
+`krt` provides a framework for building _declarative_ controllers.
+See the [design doc](https://docs.google.com/document/d/1-ywpCnOfubqg7WAXSPf4YgbaFDBEU9HIqMWcxZLhzwE/edit#heading=h.ffjmk8byb9gt) and [KubeCon talk](https://sched.co/1R2oY) for more background.
+
+The framework aims to solve a few problems with writing controllers:
+* Operate on any types, from any source. Kubernetes provides informers, but these only work on Kubernetes types read from Kubernetes objects.
+  * `krt` can accept any object type, from any source, and handle them the same.
+* Provide high level abstractions.
+  * Controller authors can write simple transformation functions from `Input` -> `Output` (with dependencies); the framework handles all the state automatically.
```

Comment:
However the most troublesome logic is this, we need translate one/some objects to others. I think no pkg could largely ease it.

Reply:
I agree with this. The library doesn't address this, nor could any really.

However, what the library does achieve is to make it so that _only_ business logic needs to be written. If you look at Istio controllers, actually only a tiny amount is about translation of objects. Most of it is about managing state and event notification. For example, look at https://github.com/istio/istio/blob/d8b21e02e754b6e93bf259ff7e226fafaa9c16d3/pilot/pkg/serviceregistry/serviceentry/store.go#L30 -- the entire file is just 150 lines of state management that is entirely obsolete by this library, and we know that logic has a bunch of bugs ( see my PR on the leaks, and many others in the past ).
Additionally, all the WorkloadInstanceHandler stuff is a bit of a mess.

With the library, we can simplify things quite a bit